src/gsumm.c

```diff
@@ -1162,3 +1167,83 @@ SEXP gprod(SEXP x, SEXP narmArg) {
   return(ans);
 }
 
+SEXP gshift(SEXP x, SEXP nArg, SEXP fillArg, SEXP typeArg) {
```

Comment:
shouldn't we prefer to re-use the code in shift.c here? otherwise it'll be a pain to always update code in both places going forward

Reply:
I can try to recycle as much as possible but there are some different things to consider about both implementations.

| Basic shift | gshift |
|---|---|
| works on data.tables/vectors/lists | operates on vectors |
| main workhouse is `memmove` | used memory may not be blocked |

Reply:
Maybe basic shift could be redirected to call gshift then?  It would have to call gforce prep function first, perhaps in a dummy way to trick gshift into working.  Then it would be common code and all existing tests of shift would cover the new gforce gshift too.
Perhaps not now. Happy to merge this PR as is, subject to integer64. A task could be created to redirect shift to gshift in future.